generator/dart/packages/google_cloud_gax/lib/src/encoding.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Utility methods for JSON encoding and decoding from [Message] objects.
+///
+/// See https://protobuf.dev/programming-guides/json/ for docs on the JSON
+/// encoding of many of these types.
+library;
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import '../common.dart';
+
+/// Decode an `int64` value.
+int? decodeInt64(Object? value) {
+  return value is String ? int.parse(value) : value as int?;
+}
+
+/// Decode a `double` value.
+double? decodeDouble(Object? value) {
+  return (value as num?)?.toDouble();
+}
```

Comment:
ProtoJSON uses strings to represent `NaN`, `Infinity` and `-Infinity`.  Consider opening a bug to handle these cases.

Reply:
will do

Reply:
https://github.com/googleapis/google-cloud-rust/issues/1808